Kiosk/Bid Fulfillment/RegistrationConfirmationViewController.swift

```diff
@@ -39,6 +49,23 @@ class RegistrationConfirmationViewController: UIViewController {
         let targetRange = ((confirmationLabel.text ?? "") as NSString).range(of: targetConditionsOfSaleText)
         if NSLocationInRange(indexOfCharacter, targetRange) {
             AppDelegate().showUserWebViewAtAddress(ConditionsOfSaleLink)
+        } else {
+            // We're going to assume that they tapped on "I agree to the" and toggle the checkmark
+            toggleConditionsOfSaleChecked()
+        }
+    }
+
+    @IBAction func conditionsOfSaleCheckmarkTapped(_ sender: Any) {
+        toggleConditionsOfSaleChecked()
+    }
+
+    func toggleConditionsOfSaleChecked() {
+        _conditionsOfSaleChecked.value = !_conditionsOfSaleChecked.value
+
+        if _conditionsOfSaleChecked.value {
+            conditionsOfSaleCheckbox.setImage(UIImage.init(named: "checkmark_checked"), for: .normal)
+        } else {
+            conditionsOfSaleCheckbox.setImage(UIImage.init(named: "checkmark_unchecked"), for: .normal)
```

Comment:
In react i'd expect the visual result to be declared separately from the effect of toggling the checkbox, but I'm guessing these are more naturally colocated in the context of a view controller.

Reply:
Yeah exactly.